lib/games_puzzles_algorithms/games/ttt/game_state.py

```diff
@@ -72,6 +72,137 @@ def __iter__(self):
         return (i for i in self._data)
 
 
+class WinDetector(object):
+    """
+    The WinDetector is used to track the win state of a game of tic-tac-toe.
+    This is done by tracking run-lengths on the board for each direction. We
+    perform this count from left to right and top to bottom along the board.
+    The value in any given cell if the length of the run ending in that cell.
+    """
+    def __init__(self, rows, columns, spaces_to_win):
```

Comment:
Names like this confuse me. For example, a **row** is a list of cells, each of which could be something like a number, a coloured stone, or a symbol. The **number of rows** is the length of this list. As such, I prefer using the prefix `num` to denote the count of objects. So `num_rows` instead of `rows`. Maybe it's just me and others can distinguish well enough based on the context, but I find it less readable. I would prefer adding the `num` prefix to these arguments and their associated member variables.